### DISCUSSION TITLE
Change Codepage and Encoding

### DISCUSSION THREAD

**Discussion by MocniMax** [jmstoolbox/jmstoolbox#109] [Q&A]
Hi,

Is it possible to select different Codepage and Encoding while displaying messages? It is particularly necessary when working on zOS MQ.

Regards

**Comment by titou10titou10** [comment]
Currently no.
..but I could add the missing MQ properties in the "properties" section of the JTB session definition

What properties do you need, those ones?
- [CCSID]( https://www.ibm.com/docs/en/ibm-mq/9.0?topic=objects-ccsid)
- [ENCODING](https://www.ibm.com/docs/en/ibm-mq/9.0?topic=objects-encoding)

Others?
[The list of MQ of properties is here](https://www.ibm.com/docs/en/ibm-mq/9.0?topic=reference-properties-mq-classes-jms-objects#jm10910_)

**Comment by MocniMax** [comment]
Yes, I meant CCSID and ENCODING. To have them as session properties should be enough because a goal is to have all message control i.e. binary stored parts readable - Message ID, Correlation ID, Group ID.
Having all of them visible as standard message data in queue tab will be very useful.

Another point is to be able to specify/change CCSID and ENCODING while sending new messages.  Both are very useful for us working with IBM MQ on AIX and zOS.

Basically, I am missing additional MQM and queue related data which are probably more administrator related. Some of them are:
- Type (Local, Remote, Transmission, Alias, ...)
- Number or uncommitted messages
- Is Get/Put allowed
- Alias target queue name for Alias queue
- Maximal number of messages
- Connection data for Remote queue - Target remote queue name, Remote MQM name, Transmission queue 

It would be nice to have that data together with queue depth in the same tab as MQM Session properties for all listed queues.

Thanks..

**Reply by titou10titou10** [reply]
Most of the information you requested are already there: in the queue tree, right click on a destination, "Destination information"

What do you mean by *"to be able to specify/change CCSID and ENCODING while sending new messages"*?
AFAIK`"CCSID"`and`"ENCODING"`are set ate the connection level, not per message...

**Reply by MocniMax** [reply]
True, I didn't realize that :( , should check all available options and dialogs before asking for something :). I would prefer to have all those data listed in one table for all available queues but can use this approach too.

Yes, CCSID and ENCODING are connection related and I often need to specify them especially when I need to transfer data from AIX to zOS and vice versa. 

BTW, how can I export and import a bulk of messages as one file? Is it possible to dump a complete queue in a single file and then load it back?
I tried Export Payload and as template but both are creating one file per selected message.

**Comment by titou10titou10** [comment]
Feature requests created CCSID: #112 ENCODING: #113 

CCSID is easy to implement as it is set at the session level
ENCODING is at the destination level (I was wrong previsouly..) and it will take much more work to implement as JTB uses generic JMS to deal with messages. The "specific" part per QMin JTB  is only when connecting

**Comment by MocniMax** [comment]
Great, thanks.
